gtsfm/evaluation/metrics.py

```diff
@@ -89,9 +89,10 @@ def __init__(
 
         self._name = name
         if data is not None:
-            # Cast to a numpy array
+            # Cast to a numpy array, unless data is an empty string,
+            # which enables blank values in metrics report.
             if not isinstance(data, np.ndarray):
-                data = np.array(data, dtype=np.float32)
```

Comment:
as mentioned above, there are some methods in this class that assume a metric is a number, and I think its better to keep it that way. it has also caused some NaN/None bugs and we should not change it unless there are good reasons to.

Reply:
Reverted.

Reply:
please revert the comment as well :)

Reply:
actually, all changes to this file? is the other one necessary?

Reply:
Reverted the comment and other change (was necessary when I was storing empty strings).